internal/fs/fs.go

```diff
@@ -2658,32 +2658,31 @@ func (fs *fileSystem) WriteFile(
 		defer cancel()
 	}
 
-	if fs.newConfig.Write.ExperimentalEnableRapidAppends {
-		fs.mu.Lock()
-		fh := fs.handles[op.Handle].(*handle.FileHandle)
-		fs.mu.Unlock()
-
-		//TODO: Initialize BWH before invoking write()
-		if _, err := fh.Write(ctx, op.Data, op.Offset); err != nil {
-			return err
-		}
-		return
-	}
-	// Find the inode.
+	// Find the inode( and file handle in case of appends).
 	fs.mu.Lock()
+	fh := fs.handles[op.Handle].(*handle.FileHandle)
 	in := fs.fileInodeOrDie(op.Inode)
 	fs.mu.Unlock()
 
-	in.Lock()
-	defer in.Unlock()
+	var gcsSynced bool
+	if fs.newConfig.Write.ExperimentalEnableRapidAppends {
+		err = fs.initBufferedWriteHandlerAndSyncFileIfEligible(ctx, in, fh.OpenMode())
```

Comment:
why can't it be always fh.openMode() ?? that way we can move this line out of if-else